packages/components/focus-trap/src/utils.ts

```diff
@@ -132,3 +141,45 @@ export const focusFirstDescendant = (
 }
 
 export const focusableStack = createFocusableStack()
+
+export const isFocusCausedByUserEvent = (): boolean => {
+  return lastUserFocusTimestamp.value > lastAutomatedFocusTimestamp.value
+}
+
+export const useFocusReason = (): {
+  focusReason: typeof focusReason
+  lastUserFocusTimestamp: typeof lastUserFocusTimestamp
+  lastAutomatedFocusTimestamp: typeof lastAutomatedFocusTimestamp
+} => {
+  if (!isFocusReasonHandlersAttached) {
+    isFocusReasonHandlersAttached = true
+    if (typeof document !== 'undefined') {
```

Comment:
Maybe the logic here can be added to `onMounted` and then remove the bound event in `onBeforeUnmount`?

Reply:
> This happens only on date/time pickers.

All inputs should be re-focused in this scenario. Date/time picker is the only one designed to always have the popup open when it is focused.

Reply:
Done